src/test/java/com/microsoft/sqlserver/jdbc/AlwaysEncrypted/MultiUserAKVTest.java

```diff
@@ -0,0 +1,643 @@
+                + ")";
+
+        try (SQLServerConnection con = PrepUtil.getConnection(connString, AEInfo);
+                SQLServerStatement stmt = (SQLServerStatement) con.createStatement()) { 
+            TestUtils.dropTableIfExists(tableName, stmt);
+            stmt.execute(sqlQuery);                              
+        } catch(SQLException e) {
+            fail(e.getMessage());
+        }        
+    }
+
+    private void dropObject(String connString, String objectType, String objectName) {
+        try (SQLServerConnection con = (SQLServerConnection) PrepUtil.getConnection(connString, AEInfo);
+                SQLServerStatement stmt = (SQLServerStatement) con.createStatement()) {
+            switch(objectType) {
+                case "TABLE":
+                    TestUtils.dropTableIfExists(objectName, stmt);
+                    break;
+                case "CEK":
+                    dropCEK(objectName, stmt);
+                    break;
+                case "CMK":
+                    dropCMK(objectName, stmt);
+                    break;
+                default:
+                    break;
+            }
+        } catch(Exception ex) {
+            fail(ex.getMessage());
+        }
+    }
+
+    private boolean cekCacheContainsKey(byte[] encryptedKey, SQLServerColumnEncryptionAzureKeyVaultProvider provider) throws Exception {
+        assertFalse(null == encryptedKey || 0 == encryptedKey.length);        
+
+        String encryptedCEKHexString = byteToHexDisplayString(encryptedKey);    
+
+        Field cekCacheField = provider.getClass().getDeclaredField(cekCacheName);
+        cekCacheField.setAccessible(true);
+
+        Object fieldValue = cekCacheField.get(provider);
+        Method method = fieldValue.getClass().getDeclaredMethod("contains", Object.class);
+        method.setAccessible(true);
+
+        boolean result = (boolean)method.invoke(fieldValue, encryptedCEKHexString);
+        
+        return result;
+    }
+
+    private byte[] signColumnMasterKeyMetadata(SQLServerColumnEncryptionAzureKeyVaultProvider provider, String masterKeyPath, boolean allowEnclaveComputations) throws Exception {
+
+        MessageDigest md = MessageDigest.getInstance("SHA-256");
+        md.update(Constants.AZURE_KEY_VAULT_NAME.toLowerCase().getBytes(java.nio.charset.StandardCharsets.UTF_16LE));
+        md.update(masterKeyPath.toLowerCase().getBytes(java.nio.charset.StandardCharsets.UTF_16LE));
+        
+        if (allowEnclaveComputations) {
+            md.update("true".getBytes(java.nio.charset.StandardCharsets.UTF_16LE));
+        } else {
+            md.update("false".getBytes(java.nio.charset.StandardCharsets.UTF_16LE));
+        }
+
+        byte[] dataToSign = md.digest();
+        if (null == dataToSign) {
+            fail("data to sign is null or empty.");

```

Comment:
this isn't defined in TestResource. Can we have just a generic {0} is null or empty error for all these cases instead of adding each separately? (same for line 589)

Reply:
Added generic message template.